modules/battery_manager/php/testcontroller.class.inc

```diff
@@ -0,0 +1,251 @@
+ * Allows users to add, activate, and deactivate entries in the test battery.
+ *
+ * PHP Version 7
+ *
+ * @category Module
+ * @package  Battery_Manager
+ * @author   Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class TestController implements RequestHandlerInterface
+{
+
+    /**
+     * Returns true if user has access to this endpoint.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return true;
+    }
+
+    /**
+     * This function passes the request to the handler. This is necessary since
+     * the Endpoint bypass the Module class.
+     *
+     * XXX: This function should be extracted to a parent class.
+     *
+     * @param ServerRequestInterface  $request The PSR7 request.
+     * @param RequestHandlerInterface $handler The request handler.
+     *
+     * @return ResponseInterface The outgoing PSR7 response.
+     */
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ) : ResponseInterface {
+        return $handler->handle($request);
+    }
+
+    /**
+     * This acts as an Ajax enpoint that handles all action requests from the
+     * Battery Manager Module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db     = \Database::singleton();
+        $user   = $request->getAttribute('user');
+        $method = $request->getMethod();
+        $test   = json_decode($request->getBody()->getContents(), true);
+
+        switch($method) {
+        case 'GET':
+            return $this->getInstances($user);
+        case 'POST':
+            return $this->addInstance($db, $user, $test);
+        case 'PUT':
+            return $this->updateInstance($db, $test);
+        }
+    }
+
+    /**
+     * Gets the data source for battery manager tests.
+     *
+     * @param \User $user User Instance
+     *
+     * @return \LORIS\Data\Provisioner
+     */
+    function getDataProvisioner(\User $user) : \LORIS\Data\Provisioner
+    {
+        $provisioner = new TestProvisioner();
+
+        if ($user->hasPermission('access_all_profiles') == false) {
+            $provisioner = $provisioner->filter(
+                new \LORIS\Data\Filters\UserSiteMatch()
+            );
+        }
+        return $provisioner;
+    }
+
+    /**
+     * Updates the status of the given test from Active to Deactive or
+     * viceversa.
+     *
+     * @param \Database $db   Database Instance
+     * @param array     $data Data to be saved
+     *
+     * @return \LORIS\Http\Response response
+     */
+    function updateInstance(\Database $db, array $data)
```

Comment:
Return type should be added to the signature. 
On a side note, I would make all the functions private except those that are part of the `RequestHandlerInterface`. Main reasons are that it is easier to change a function signature when it is only used inside one class. And making a private function public is easier than making a public function private.

Reply:
Okay

Reply:
given the phpcs standards this requires all private functions to be prepended with `_`